### PR TITLE
215 2 bundle size

### PR DESCRIPTION
**Pull request by KyleW**
This simplifies the gulpfile, fixes a variety of lint errors, adds ng-annotate and most importantly turns uglify back on.

This takes the app bundle size from 5.5mb to 2.8mb.

### CODE REVIEW

**Comment by KyleW** [comment]
I've made some progress. I've got minification and sourcemaps working and removed at least one redundant dependency. The bundle is down to 2.3mb before gzipping (from 5.5mb). This is worth merging as is.

However, browserify is slow! That's the new problem. Next step is to split out the vendor files into a separate bundle. And some steps to make builds faster in dev mode (incremental builds, turning off minification in dev, etc.). I'll wrap that work up in a separate pull request.

File: app/vendor/vendor.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ var angularSanitize = require('angular-sanitize');
 var angularBreadcrumb = require('angular-breadcrumb');
 var bootstrap = require('bootstrap');
 var domReady = require('domready');
-var AgGrid = require('ag-grid');
+var AgGrid = require('../../node_modules/ag-grid/dist/ag-grid.noStyle');
```
Comment (tdooner) [inline app/vendor/vendor.js]: Huh. Did it not resolve as just `require('ag-grid.noStyle')`?

Comment (tdooner) [inline app/vendor/vendor.js]: Oh I see, you copied this from somewhere else.

File: gulp.config.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        //    ]
+        //},
+        copyIndex: {
+            src: './app/index.html',
+            watch: './app/index.html'
+        },
+        copyFavicon: {
+            src: './app/favicon.png'
+        }
+    };
```
Comment (tdooner) [inline gulp.config.js]: 4 space indents? 😬

**Comment by tdooner** [comment]
Nice work! Probably @adborden should look at it too, but anyway this seems like a great step forward to me.

File: app/vendor/vendor.js (hunk 1 above)
Comment (adborden) [inline app/vendor/vendor.js]: Can this just be `require('ag-grid/dist/ag-grid.noStyle')`?

File: gulp.config.js (hunk 2 above)
Comment (adborden) [inline gulp.config.js]: I'm seeing a mix of 2 space and 4 space, is there something misconfigured in our lint config?

File: gulpfile.js [hunk 3]
```diff
@@ -151,158 +97,111 @@ gulp.task('clean-dev', function() {
...
+        .transform(markedify)
+        .bundle() // Start bundle
+        .pipe(source(filePath.browserify.src)) // Entry point
+        .pipe(buffer()) // Convert to gulp pipeline
+        .pipe(sourceMaps.init({
+            loadMaps: true
+        })) // Strip inline source maps
+            .pipe(ngAnnotate())
+            .pipe(uglify({
+                mangle: false
```
Comment (adborden) [inline gulpfile.js]: Should we be doing a `mangle: true` as part of the `build-prod` task? That would get us some extra savings on production, right?

**Comment by adborden** [comment]
Also sorry it took me a while to get to this :)

File: app/vendor/vendor.js (hunk 1 above)
Comment (KyleW) [inline app/vendor/vendor.js]: Probably. But I'm actually thinking of removing ag-grid entirely. It seems like we don't do a whole lot with it and it adds several hundred kb of js. Either way, I'll do something about this in the next pr.

File: gulp.config.js (hunk 2 above)
Comment (KyleW) [inline gulp.config.js]: ah, sorry, my editor was in work mode. Will clean that up.

File: gulpfile.js (hunk 3 above)
Comment (KyleW) [inline gulpfile.js]: The total bundle is down to 2.33mb. Here's the biggest contributions by size after minification:
ag-grid: 585kb (a good candidate for elimination)
angular: 279kb
d3: 237kb (are we using this anywhere?)
angular-ui-bootstrap: 171kb
jquery: 129kb (?)
lodash: 124kb
swagger-client: 70kb (should be commented out at least)
appcode: 121kb

Maybe my next chunk of work will be removing some of these.

Comment (adborden) [inline gulpfile.js]: > d3: 237kb (are we using this anywhere?)

Long ago some folks were playing with visualizations but nothing is implemented, we can remove this.

> angular-ui-bootstrap: 171kb

I propose we remove this, I think the plan was to use more components from ui-bootstrap but I don't think it's been really worth it. Just my 2¢

> jquery: 129kb (?)

This is a dependency for some of the bootstrap/ui-bootstrap components. I think we're at least using the nav on mobile which requires this, but I would also be happy to come up with a bespoke solution if it means dropping ui-bootstrap and jquery.

> lodash: 124kb

I'd also be happy to drop this. IIRC, theres a handful of array iteration functions we're using that can just be replaced with an ES5/ES6 shim.

> swagger-client: 70kb (should be commented out at least)

Yep, I feel like we won't return to it anytime soon. I'd like to remove this as part of a PR removing `disclosureApi`. Happy to take that on unless you want it.

**Pull request merged**